client/zanata-client-commands/src/main/java/org/zanata/client/commands/PutUserCommand.java

```diff
@@ -21,22 +22,45 @@ public PutUserCommand(PutUserOptions opts) {
     }
 
     public void run() throws Exception {
-        Account account = new Account();
-        account.setEmail(getOpts().getUserEmail());
-        account.setName(getOpts().getUserName());
-        account.setUsername(getOpts().getUserUsername());
-        account.setPasswordHash(getOpts().getUserPasswordHash());
-        account.setApiKey(getOpts().getUserKey());
+
+        String userUsername = getOpts().getUserUsername();
+        String passwordHash = getOpts().getUserPasswordHash();
+        String userEmail = getOpts().getUserEmail();
+
+        Account account = getClientFactory().getAccountClient().get(userUsername);
+
+        if (account == null) {
+            log.info("Creating new account {}", userUsername);
+            account = new Account();
+            if (userEmail == null || userUsername == null) {
+                throw new RuntimeException("New user's name and email must be specified");
+            } else if (StringUtils.isBlank(passwordHash)) {
+                log.warn("Unusable passwordHash set, user will need to use the Forgot Password feature");
+                account.setPasswordHash("NULL");
```

Comment:
> I can't pass an empty string, it gets blocked somewhere I don't understand, and then liquibase replaces empties with null.

Probably caused by the `@NonEmpty` annotation on the DTO. When that happens, do the nulls in the DB cause a problem?

Anyway, I think we'd better change the API to remove `@NonEmpty` and `required = true` from the passwordHash field, plus ensure the server does the right thing with a blank passwordHash.

It's not even a required field in the DB, from what I can tell, which means the server (when GETting an Account) can't currently generate a valid DTO from the HAccount in the DB if the passwordHash is blank.